lib/ex_doc/retriever.ex

```diff
@@ -15,13 +15,21 @@ defmodule ExDoc.Retriever do
   """
   @spec docs_from_dir(Path.t() | [Path.t()], ExDoc.Config.t()) :: [ExDoc.ModuleNode.t()]
   def docs_from_dir(dir, config) when is_binary(dir) do
-    pattern = if config.filter_prefix, do: "Elixir.#{config.filter_prefix}*.beam", else: "*.beam"
-    files = Path.wildcard(Path.expand(pattern, dir))
-    docs_from_files(files, config)
+    dir
+    |> __docs_from_dir__(config)
+    |> reject_docs_with_moduledoc_false()
   end
 
   def docs_from_dir(dirs, config) when is_list(dirs) do
-    Enum.flat_map(dirs, &docs_from_dir(&1, config))
+    dirs
+    |> Enum.flat_map(&__docs_from_dir__(&1, config))
+    |> reject_docs_with_moduledoc_false()
+  end
+
+  defp __docs_from_dir__(dir, config) do
```

Comment:
Those changes do not seem to be necessary?

Reply:
fixed